### PR TITLE
Don't call release() before retain() #13656

### PR DESCRIPTION
**Pull request by seivan**
My c++ is pretty poor, but it might help with https://github.com/cocos2d/cocos2d-x/issues/13656

### CODE REVIEW

**Comment by WenhaiLin** [comment]
@seivan This is not correct solution for issue #13656.  please review suggestions from @c3076802 .

**Comment by seivan** [comment]
Squashed and updated with the fix from @c3076802

**Comment by WenhaiLin** [comment]
[ci rebuild]

File: cocos/base/CCMap.h [hunk 1]
```diff
@@ -274,9 +274,9 @@ class Map
     void insert(const K& key, V object)
     {
         CCASSERT(object != nullptr, "Object is nullptr!");
-        erase(key);
-        _data.insert(std::make_pair(key, object));
         object->retain();
+        erase(key)
```
Comment (WenhaiLin) [inline cocos/base/CCMap.h]: You should add a ";" (semicolon) at the end.

**Comment by WenhaiLin** [comment]
@pandamicro I think this PR is OK for merge.

**Pull request merged**